client/src/pages/UserPasswordChangeRequired.tsx

```diff
@@ -0,0 +1,96 @@
+import { useState } from 'react';
+import { Form, Button, Alert } from 'react-bootstrap';
+import { UserPasswordChangeDTO, UserService } from '../api/user.api';
+import { useNavigate } from 'react-router-dom';
+import { AxiosError } from 'axios';
+
+export const UserPasswordChangeRequired = () => {
+    let navigate = useNavigate();
+    const [id, setId] = useState(0);
+    const [oldPassword, setOldPassword] = useState('');
+    const [newPassword, setNewPassword] = useState('');
+    const [confirmNewPassword, setConfirmNewPassword] = useState('');
+    const [error, setError] = useState('');
+
+    const handleSubmit = (e: React.FormEvent) => {
+        e.preventDefault();
+        if (newPassword !== confirmNewPassword) {
+            setError('New password and new password confirmation do not match');
+            return;
+        }
+
+        let dto: UserPasswordChangeDTO = {
+            id: id,
+            old_password: oldPassword,
+            new_password: newPassword
+        };
+
+        UserService.ChangePassword(dto).then((_) => {
+            console.log("TODO: Remove JWT from browser storage");
+            navigate("/login", { replace: true });
+        }).catch((err: AxiosError) => {
```

Comment:
Here, the user can see all the details packed into the value of the `message `attribute.